backends/gstreamer/lib.rs

```diff
@@ -139,6 +140,29 @@ impl GStreamerBackend {
             backend_chan: Arc::new(Mutex::new(backend_chan)),
         }))
     }
+
+    fn media_instance_action(
+        &self,
+        id: &ClientContextId,
+        cb: Rc<dyn Fn(Arc<Mutex<dyn MediaInstance>>) -> Result<(), ()>>,
```

Comment:
It's not clear why the Rc is needed here - what if we use `&dyn Fn(...) -> ...` instead, and don't add `move` to each closure? Also, why not call `lock().unwrap()` in this method and pass `&mut dyn MediaInstance` to the callback instead?